internal/bigquery/batch/client.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2021 OTA Insight Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package batch
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/OTA-Insight/bqwriter/internal"
+
+	"cloud.google.com/go/bigquery"
+)
+
+var (
+	autoDetectAndSchemaNotSupported = errors.New("BQ batch client: usage of autodetect being true and schema being passed is not supported")
```

Comment:
Can you end all these errors on `Err`, so far that's the convention I've taken with this library